go.mod

```diff
@@ -3,92 +3,102 @@ module github.com/openshift/hive
 	github.com/terraform-providers/terraform-provider-aws => github.com/openshift/terraform-provider-aws v1.60.1-0.20200630224953-76d1fb4e5699
 	github.com/terraform-providers/terraform-provider-azurerm => github.com/openshift/terraform-provider-azurerm v1.40.1-0.20200707062554-97ea089cc12a
 	github.com/terraform-providers/terraform-provider-ignition/v2 => github.com/community-terraform-providers/terraform-provider-ignition/v2 v2.1.0
-	sigs.k8s.io/cluster-api-provider-aws => github.com/openshift/cluster-api-provider-aws v0.2.1-0.20200506073438-9d49428ff837
-	sigs.k8s.io/cluster-api-provider-azure => github.com/openshift/cluster-api-provider-azure v0.1.0-alpha.3.0.20200120114645-8a9592f1f87b
-	sigs.k8s.io/cluster-api-provider-openstack => github.com/openshift/cluster-api-provider-openstack v0.0.0-20200526112135-319a35b2e38e
+	kubevirt.io/client-go => kubevirt.io/client-go v0.29.0
+	sigs.k8s.io/cluster-api-provider-aws => github.com/openshift/cluster-api-provider-aws v0.2.1-0.20201022175424-d30c7a274820
+	sigs.k8s.io/cluster-api-provider-azure => github.com/openshift/cluster-api-provider-azure v0.1.0-alpha.3.0.20201016155852-4090a6970205
+	sigs.k8s.io/cluster-api-provider-openstack => github.com/openshift/cluster-api-provider-openstack v0.0.0-20201116051540-155384b859c5
 )
 
-// needed because otherwise v12.0.0 is picked up as a more recent version
-replace k8s.io/client-go => k8s.io/client-go v0.19.0
-
-// needed because otherwise installer fetches a library-go version that requires bitbucket.com/ww/goautoneg which is dead
-// Tagged version fetches github.com/munnerz/goautoneg instead
-replace github.com/openshift/library-go => github.com/openshift/library-go v0.0.0-20200918101923-1e4c94603efe
+// from installer
+// Prevent the following modules from upgrading version as result of terraform-provider-kubernetes module
+// The following modules need to be locked to compile correctly with terraform-provider-azure and terraform-provider-google
+replace (
+	github.com/Azure/go-autorest/autorest/adal => github.com/Azure/go-autorest/autorest/adal v0.8.2
+	github.com/Azure/go-autorest/autorest/azure/cli => github.com/Azure/go-autorest/autorest/azure/cli v0.3.1
+	github.com/Azure/go-autorest/autorest/date => github.com/Azure/go-autorest/autorest/date v0.2.0
+	github.com/Azure/go-autorest/autorest/validation => github.com/Azure/go-autorest/autorest/validation v0.2.1-0.20191028180845-3492b2aff503
+	github.com/apparentlymart/go-cidr => github.com/apparentlymart/go-cidr v1.0.1
+	github.com/go-openapi/spec => github.com/go-openapi/spec v0.19.0
+	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1
+	github.com/hashicorp/go-plugin => github.com/hashicorp/go-plugin v1.2.2
+	github.com/ulikunitz/xz => github.com/ulikunitz/xz v0.5.7
+	google.golang.org/api => google.golang.org/api v0.25.0
+	google.golang.org/genproto => google.golang.org/genproto v0.0.0-20200526211855-cb27e3aa2013
+	google.golang.org/grpc => google.golang.org/grpc v1.29.1
+)
```

Comment:
This is way better - I wonder if this warranties some changes [here](https://github.com/openshift/hive/blob/master/docs/developing.md#vendoring-the-openshift-installer)

Reply:
this is one off change we need to replace because of https://github.com/openshift/installer/pull/4350 , i think this should be removed when installer bumps to set of dependencies that don't need this specifically.

the doc provides general guidelines of how to add or update the dependencies in most/usual case, like the cluster-api vendor bumps for aws etc included here too in the original replace block